packages/dosespot-react/src/DoseSpotNewMedicationForm.tsx

```diff
@@ -0,0 +1,154 @@
+import { Box, Button, Divider, Group, NumberInput, Stack, Text, TextInput } from '@mantine/core';
+import { NDC, RXNORM } from '@medplum/core';
+import { Bundle, MedicationKnowledge } from '@medplum/fhirtypes';
+import React, { useState } from 'react';
+import { DOSESPOT_DISPENSABLE_DRUG_ID_SYSTEM, DOSESPOT_REFILLS_SYSTEM } from './common';
+import { DoseSpotMedicationSelect } from './DoseSpotMedicationSelect';
+
+export interface DoseSpotNewMedicationFormProps {
+  readonly searchMedications: (term: string) => Promise<Bundle<MedicationKnowledge> | undefined>;
+  readonly addFavoriteMedication: (medication: MedicationKnowledge) => Promise<void>;
+  readonly loading?: boolean;
+}
+
+interface FormData {
+  quantity: number;
+  refills: number;
+  directions: string;
+}
+
+export function DoseSpotNewMedicationForm(props: DoseSpotNewMedicationFormProps): React.JSX.Element {
+  const { searchMedications, addFavoriteMedication, loading = false } = props;
+  const [formData, setFormData] = useState<FormData>({
+    quantity: 1,
+    refills: 0,
+    directions: '',
+  });
+  const [medication, setMedication] = useState<MedicationKnowledge | null>(null);
+
+  const handleAddFavorite = async (): Promise<void> => {
+    // Create the updated medication with proper FHIR structure
+    const completedMedication: MedicationKnowledge = {
+      ...medication,
+      resourceType: 'MedicationKnowledge',
+      // Use amount field for quantity
+      amount: {
+        value: formData.quantity,
+        unit: 'units',
+      },
+      extension: [
+        {
+          url: DOSESPOT_REFILLS_SYSTEM,
+          valueString: formData.refills.toString(),
+        },
+      ],
+      // Add administration guidelines for directions
+      administrationGuidelines: [
+        {
+          dosage: [
+            {
+              dosage: [
+                {
+                  patientInstruction: formData.directions,
+                },
+              ],
+              type: {
+                coding: [
+                  {
+                    system: 'https://dosespot.com/patient-instructions',
+                  },
+                ],
+              },
+            },
+          ],
+        },
+      ],
+    };
+
+    await addFavoriteMedication(completedMedication);
+    setMedication(completedMedication);
+  };
+
+  const getCodingValue = (medication: MedicationKnowledge, system: string): string | undefined => {
```

Comment:
We have a utility function for this: [`getCodeBySystem`](https://www.medplum.com/docs/sdk/core.getcodebysystem)